apps/src/javalab/CommitDialog.jsx

```diff
@@ -2,49 +2,58 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import i18n from '@cdo/javalab/locale';
 import color from '@cdo/apps/util/color';
-import {CompileStatus} from './constants';
 import StylizedBaseDialog, {
   FooterButton
 } from '@cdo/apps/componentLibrary/StylizedBaseDialog';
+import {connect} from 'react-redux';
 
-export default class CommitDialog extends React.Component {
+class CommitDialog extends React.Component {
   state = {
     filesToBackpack: [],
-    commitNotes: null,
-    showCompileStatus: false
+    commitNotes: '',
+    saveInProgress: false,
+    hasSaveError: false
   };
 
   renderFooter = () => {
-    let compileStatusContent = '';
+    let footerIcon = '';
+    let footerMessageTitle = '';
+    let footerMessageText = '';
     let commitText = i18n.commit();
-    const isCommitButtonDisabled = !this.state.commitNotes;
+    const isCommitButtonDisabled =
+      !this.state.commitNotes || this.state.saveInProgress;
     if (this.state.filesToBackpack.length > 0) {
       commitText = i18n.commitAndSave();
     }
-    if (this.state.showCompileStatus) {
-      switch (this.props.compileStatus) {
-        case CompileStatus.LOADING:
-          compileStatusContent = i18n.compiling();
-          break;
-        case CompileStatus.ERROR:
-          compileStatusContent = i18n.compileFailed();
-          break;
-      }
+
+    // TODO: Add compile status here
+    if (this.state.saveInProgress) {
+      footerIcon = (
+        <span className="fa fa-spin fa-spinner" style={styles.spinner} />
+      );
+      footerMessageTitle = i18n.saving();
+    } else if (this.state.hasSaveError) {
+      footerIcon = (
+        <span className="fa fa-exclamation-circle" style={styles.iconError} />
+      );
+      footerMessageTitle = i18n.backpackSaveErrorTitle();
+      footerMessageText = i18n.backpackSaveErrorMessage();
```

Comment:
Would there be an easy way to pipe through which files didn't save here from fileUploadsFailed? Or is this part of other error messaging tasks for the future?

Reply:
We don't have a ton of real estate on the modal for error messaging, and it's not a big deal if they re-try the same list of files again. I think for now we can leave as is and add that in the future (it wouldn't be hard to pipe through, it's more a matter of where to put it).